packages/common/src/services/audius-backend/AudiusBackend.ts

```diff
@@ -2069,6 +2073,303 @@ export const audiusBackend = ({
     }
   }
 
+  function getDiscoveryEntityType(type: string) {
+    if (type === 'track') {
+      return Entity.Track
+    }
+    return Entity.Playlist
+  }
+
+  function formatBaseNotification(notification: DiscoveryNotification) {
+    const timestamp = notification.actions[0].timestamp
+    return {
+      groupId: notification.group_id,
+      timestamp: new Date(timestamp * 1000).toString(),
+      isViewed: notification.seen_at == null,
+      id: `timestamp:${timestamp}:group_id:${notification.group_id}`
+    }
+  }
+
+  function mapDiscoveryNotification(notification: DiscoveryNotification) {
+    console.log({ notification })
+    if (notification.type === 'follow') {
+      const userIds = notification.actions.map((action) => {
+        const data = action.data
+        return decodeHashId(data.follower_user_id)
+      })
+      return {
+        type: NotificationType.Follow,
+        userIds,
+        ...formatBaseNotification(notification)
+      }
+    } else if (notification.type === 'repost') {
+      let entityId
+      let entityType
+      const userIds = notification.actions.map((action) => {
+        const data = action.data
+        entityId = decodeHashId(data.repost_item_id)
```

Comment:
im a little confused with the let + map for entityId and entitiyType, does it just become the values of the last action?